x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -584,9 +590,17 @@ protected QueryTranslation asQuery(In in, boolean onAggs) {
             else {
                 Query q = null;
                 if (in.value() instanceof FieldAttribute) {
-                    FieldAttribute fa = (FieldAttribute) in.value();
                     // equality should always be against an exact match (which is important for strings)
-                    q = new TermsQuery(in.source(), fa.exactAttribute().name(), in.list());
+                    FieldAttribute fa = (FieldAttribute) in.value();
+                    List<Expression> list = in.list();
+                    // TODO: this needs to be handled inside the optimizer
+                    list.removeIf(e -> DataTypes.isNull(e.dataType()));
+                    DataType dt = list.get(0).dataType();
+                    Set<Object> set = new LinkedHashSet<>(CollectionUtils.mapSize(list.size()));
+                    for (Expression e : list) {
+                        set.add(SqlDataTypeConverter.convert(e.fold(), dt));
```

Comment:
one of the tests fails seems because of addition of SqlDataTypeConverter.convert